packages/wdio-browserstack-service/src/Percy/Percy.ts

```diff
@@ -0,0 +1,174 @@
+import fs from 'node:fs'
+import path from 'node:path'
+import os from 'node:os'
+
+import { createRequire } from 'node:module'
+const require = createRequire(import.meta.url)
+const { spawn } = require('node:child_process')
+
+import { nodeRequest, getBrowserStackUser, getBrowserStackKey, sleep } from '../util.js'
+import { PercyLogger } from './PercyLogger.js'
+
+import PercyBinary from './PercyBinary.js'
+
+import type { BrowserstackConfig, UserConfig } from '../types.js'
+import type { Options } from '@wdio/types'
+
+const logDir = 'logs'
+
+class Percy {
+    #logfile: string = path.join(logDir, 'percy.log')
+    #address: string = process.env.PERCY_SERVER_ADDRESS || 'http://127.0.0.1:5338'
+
+    #binaryPath: string | any = null
+    #options: BrowserstackConfig & Options.Testrunner
+    #config: Options.Testrunner
+    #proc: any = null
+    #isApp: boolean = false
+    #projectName: string | undefined = undefined
+
+    isProcessRunning = false
+
+    constructor(options: BrowserstackConfig & Options.Testrunner, config: Options.Testrunner, bsConfig: UserConfig) {
+        this.#options = options
+        this.#config = config
+        if (options.app) {
+            this.#isApp = true
+        }
+        this.#projectName = bsConfig.projectName
+    }
+
+    async #getBinaryPath(): Promise<string> {
+        if (!this.#binaryPath) {
+            const pb = new PercyBinary()
+            this.#binaryPath = await pb.getBinaryPath(this.#config)
+        }
+        return this.#binaryPath
+    }
+
+    async healthcheck() {
+        try {
+            const resp = await nodeRequest('GET', 'percy/healthcheck', null, this.#address)
```

Comment:
It would be great if BS could develop a Node.js API binding like [Sauce Labs](https://www.npmjs.com/package/saucelabs) has done so we don't need to code it here and there would be type support for responses

Reply:
@christian-bromann This is already in our roadmap. Thanks for pointing this out. However, I assume this is not a blocker for releasing this. Once, we have the bindings in place, we ensure that we will refactor these calls with the API bindings.

Reply:
Sounds good to me 👍